### ISSUE TITLE
Jira doesn't use db host from dbconfig.yml

### ISSUE THREAD

**Issue by Forbzy** [atlassian-labs/data-center-helm-charts#254]
**Describe the bug**
Jira seems to be using its own IP address for the pod instead of using the hostname defined in the dbconfig.yml
I created a mysql database using helm. I have a kubernetes service called mysql which exposes the traffic on the mysql pod.

Below is my dbconfig.yml

```
<?xml version="1.0" encoding="UTF-8"?>

<jira-database-config>
  <name>defaultDS</name>
  <delegator-name>default</delegator-name>

  <schema-name>public</schema-name>
  <database-type>mysql8</database-type>
  <jdbc-datasource>
    <url>jdbc:mysql://mysql:3306/myDatabase?useUnicode=true&amp;characterEncoding=UTF8&amp;sessionVariables=default_storage_engine=InnoDB</url>
    <username>admin</username>
    <password>admin123</password>
    <driver-class>com.mysql.cj.jdbc.Driver</driver-class>

    <pool-min-size>20</pool-min-size>
    <pool-max-size>100</pool-max-size>
    <pool-min-idle>10</pool-min-idle>
    <pool-max-idle>20</pool-max-idle>

    <pool-max-wait>30000</pool-max-wait>
    <validation-query>select 1</validation-query>
    <time-between-eviction-runs-millis>30000</time-between-eviction-runs-millis>
    <min-evictable-idle-time-millis>5000</min-evictable-idle-time-millis>
    <pool-remove-abandoned>true</pool-remove-abandoned>
    <pool-remove-abandoned-timeout>300</pool-remove-abandoned-timeout>
    <pool-test-while-idle>true</pool-test-while-idle>
    <pool-test-on-borrow>false</pool-test-on-borrow>
  </jdbc-datasource>
</jira-database-config>
```
Below is the error log.

```
Caused by: java.sql.SQLSyntaxErrorException: SELECT command denied to user 'admin'@'10.1.1.163' for table 'propertyentry'
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeQuery(ClientPreparedStatement.java:1003)
        at org.apache.commons.dbcp2.DelegatingPreparedStatement.executeQuery(DelegatingPreparedStatement.java:83)
        at com.querydsl.sql.AbstractSQLQuery.fetch(AbstractSQLQuery.java:446)
```

I've tried the fqdn below and got the same results.

mysql.NAMESPACE.svc.cluster.local:3306
mysql-0.mysql.NAMESPACE.svc.cluster.local:3306

I've also tried the below drivers.

com.mysql.jdbc.Driver
com.mysql.cj.jdbc.Driver

**Steps to Reproduce**
1. Configure Jira via helm with image atlassian/jira-software:8.18.1-jdk11
2. Configure mysql via helm from bitnami

**Expected Behaviour**
Jira connects to the mysql database.

**Comment by nanux** [comment]
This seems more like a permission issue. Jira can connect to the database but `admin` user doesn't have access to the table. I would recommend following [Jira MySQL documentation](https://confluence.atlassian.com/adminjiraserver/connecting-jira-applications-to-mysql-8-0-1018775461.html#ConnectingJiraapplicationstoMySQL8.0-configuringmysql) and create a database and a specific jira user ahead of the helm installation. Was `myDatabase` created explicitly for jira? You can check permissions with:

```
SHOW GRANTS FOR 'admin'@'localhost';
select user from mysql.db where db='myDatabase';
```

You should be able to use the MySQL credentials to connect to the database from a normal MySQL client ([example](https://github.com/docker-library/docs/blob/master/mysql/README.md#connect-to-mysql-from-the-mysql-command-line-client)).

**Comment by Forbzy** [comment]
I think you're right in that its a permissions issue. I'm using the mysql chart provided by bitnami. I've tried changing the permissions for the users. No effect so far. Below are the different commands I've tried.

```
CREATE USER 'admin'@'%' IDENTIFIED BY 'admin123';
CREATE USER 'admin'@'10.1.1.%' IDENTIFIED BY 'admin123';
CREATE USER 'admin'@'jira-0' IDENTIFIED BY 'admin123';

GRANT ALL PRIVILEGES ON jiradb.* TO 'admin'@'%' WITH GRANT OPTION;
GRANT ALL PRIVILEGES ON jiradb.* TO 'admin'@'10.1.1.%' WITH GRANT OPTION;
GRANT ALL PRIVILEGES ON jiradb.* TO 'admin'@'jira-0' WITH GRANT OPTION;
```

I always make sure to run the below as well.
`flush privileges;`

Below is my mysql configuration.

```
[mysqld]
default_authentication_plugin=mysql_native_password
skip-name-resolve
explicit_defaults_for_timestamp
basedir=/opt/bitnami/mysql
plugin_dir=/opt/bitnami/mysql/plugin
port=3306
socket=/opt/bitnami/mysql/tmp/mysql.sock
datadir=/bitnami/mysql/data
tmpdir=/opt/bitnami/mysql/tmp
max_allowed_packet=512M
bind-address=0.0.0.0
pid-file=/opt/bitnami/mysql/tmp/mysqld.pid
log-error=/opt/bitnami/mysql/logs/mysqld.log
character-set-server=UTF8MB4
collation-server=utf8mb4_bin
default-storage-engine=INNODB
innodb_default_row_format=DYNAMIC
innodb_log_file_size=2G
interactive_timeout=150000
wait_timeout=150000
```

**Comment by bianchi2** [comment]
@Forbzy  have you tried creating a dedicated user after MySQL chart gets deployed (as per [the following instructions](https://confluence.atlassian.com/adminjiraserver/connecting-jira-applications-to-mysql-8-0-1018775461.html#ConnectingJiraapplicationstoMySQL8.0-configuringmysql)), i.e. not rely on the user that gets created by Bitnami chart?

**Comment by Forbzy** [comment]
@bianchi2 I have. The strange thing about this is it was working the other day.  I havent changed any configurations. I've even tried root user.

**Comment by badgersow** [comment]
I suggest to get Jira out of the equation and try to connect to the DB directly from the Pod. Something like this:

1. Connect to the pod
```
kubectl exec -it <POD_NAME> -- bash
```

2. Install the mysql client on the pod
```
apt update; apt install mysql-client
```

3. Try to connect to the database from the pod using the client
```
mysql -h <HOST_WITH_DB> -u admin -Padmin123
```

If this connection doesn't work as well, then we definitely know that something is wrong with the DB or network Configuration. More likely DB configuration, because the error looks like permission issue.

On the other hand, if the direct connection works, but Jira keeps acting, we will know that the issue is related to the Jira.

---

I remember during the testing I have provisioned Confluence with the MySQL from bitnami chart, and the product was able to connect to the database. To be honest, I don't recall if I created a separate user from the product, or used the admin.

**Comment by bianchi2** [comment]
I used Bitnami chart as well. @badgersow  good point, let's just use mysql client to connect and see what's going on.

**Comment by Forbzy** [comment]
I managed to get it working. It looks like there is a compatibility issue between some mysql drivers and jira docker images. I was using images tagged with ubuntu like 8.18.1-ubuntu-jdk11 with mysql driver mysql-connector-java_8.0.26-1ubuntu20.04_all.deb. They were both 20.04 ubuntu so I thought they would work. In the end I found docker images 8.13.9-jdk11 and 8.18.1-jdk11 worked with mysql driver mysql-connector-java_8.0.26-1debian10_all.deb.
